tests/tests.rs

```diff
@@ -106,6 +137,50 @@ fn run_iv(group: &str, name: &str, path: &str, input: &[u8], output_ext: &str) {
     .unwrap();
 }
 
+/// Removes any snapshots that don't have a corresponding test associated with
+/// it
+fn remove_orphan_snapshots(path: &'static str, names: &[String]) {
+  let Ok(entries) = fs::read_dir(path) else {
+    println!("Cannot read path: {}", path);
+    return;
+  };
+
+  let mut orphans = false;
+  for entry in entries {
+    let Ok(entry) = entry else {
+      continue;
+    };
+
+    let path = entry.path();
+    let Some(file_name) = path.file_stem() else {
+      continue;
+    };
+
+    let Some(file_name_str) = file_name.to_str() else {
+      continue;
+    };
+
+    let file_name = file_name_str.to_owned();
+    if !names.contains(&file_name) {
+      orphans = true;
```

Comment:
When there are orphaned snapshots, instead of always deleting the files and always failing, can you branch on `should_write_snapshot()` and either delete the snapshots and succeed, or fail and leave the snapshots be?